jscomp/ml/runtime_representation.ml

```diff
@@ -0,0 +1,242 @@
+let extract_concrete_typedecl :
+    (Env.t -> Types.type_expr -> Path.t * Path.t * Types.type_declaration) ref =
+  ref (Obj.magic ())
+
+type 'value value = Known of 'value | Unknown
+
+type object_property = {
+  key: string;
+  value: runtime_js_value list value;
+  optional: bool;
+}
+and runtime_js_value =
+  | String of {value: string value}
```

Comment:
StringLiteral etc

Reply:
I opted to let `value` control whether it's a literal or not, but it's probably clearer to make actual literals. I'll revise.